src/lib.rs

```diff
@@ -170,12 +171,27 @@ pub unsafe extern "C" fn free_execution_result(result: *mut ExecutionResult) {
 
 #[repr(C)]
 pub enum ExecutionResult {
-    Handle(*mut ResultHandle),
+    Success(ExecutionData),
     Error(*mut c_char),
 }
 
+#[repr(C)]
+pub struct ExecutionData {
+    pub execution_duration_microseconds: c_uint,
+    pub handle: *mut ResultHandle,
+}
+
+impl From<qcs::ExecutionData> for ExecutionData {
+    fn from(data: qcs::ExecutionData) -> Self {
+        ExecutionData {
+            execution_duration_microseconds: data.duration.unwrap_or_default().as_micros() as c_uint,
```

Comment:
Worth using `c_ulonglong` to avoid more truncation from `u128`? It's still possibly truncating down to `u64`... but `Duration::from_micros` takes a `u64`, so it's maybe just more compatible across the boundary.

Reply:
FWIW, we use 64-bit unsigned int in other places for this value too.

Reply:
Good call, done. Thanks!